### PR TITLE
FIX creating sample data dci-cargo sampel A

### PR DESCRIPTION
**Pull request by sbkhuis**
Hi,

I was going through the samples and discovered that the sample data of dci-cargo was not created at startup, which resulted in a null pointer exception.

This fixes the problem.

By the way, I'm not able to run the forum, rental, and dci-cargo sample B examples. They all give errors on assembling/activating the application. I didn't manage to fix these errors.

Cheers,

Stijn

### CODE REVIEW

File: samples/dci-cargo/dcisample_a/src/main/java/org/qi4j/sample/dcicargo/sample_a/bootstrap/sampledata/BaseDataService.java [hunk 1]
```diff
@@ -146,6 +147,12 @@ public void createBaseData()
             HANDLING_EVENTS = uow.newEntity( HandlingEventsEntity.class, HandlingEventsEntity.HANDLING_EVENTS_ID );
 
             logger.debug( "BASIC DATA CREATED" );
+            uow.complete();
+            }
+            catch(Exception e) {
+            	uow.discard();
+            	logger.error("CANNOT CREATE BASIC DATA");
+            }
```
Comment (eskatos) [inline samples/dci-cargo/dcisample_a/src/main/java/org/qi4j/sample/dcicargo/sample_a/bootstrap/sampledata/BaseDataService.java:159]: Good catch!

Could you please indent the code properly and let the exception raise. If basic data cannot be created the program should fail.

**Comment by sbkhuis** [comment]
No problem. I will fix indent and exception raise.

**Comment by ruud** [comment]
Sherlock Stijn

**Comment by sbkhuis** [comment]
Hi eskatos,

I hope this is better indenting... But there are still errors on constructing the Qi4japplication in Sample B:

Caused by: org.qi4j.api.common.ConstructionException: [Module INFRASTRUCTURE-Indexing] Non-optional @Service interface org.qi4j.api.value.ValueSerializer was null in org.qi4j.index.rdf.query.RdfQueryParserFactory$RdfQueryParserFactoryMixin
    at org.qi4j.runtime.injection.DependencyModel.inject(DependencyModel.java:293)

Also in Forum and Rental. Do you have time to take a look at that?

Thanks,

Stijn

**Comment by eskatos** [comment]
Merged, thanks!
Yes, I'll take a look.

**Pull request merged**